File: RB101_Programming_Foundations/small_problems/Easy5/Easy5_Python/after_midnight2.py
# As seen in the previous exercise, the time of day can be represented as the number of minutes before or after midnight. If the number of minutes is positive, the time is after midnight. If the number of minutes is negative, the time is before midnight.

# Write two methods that each take a time of day in 24 hour format, and return the number of minutes before and after midnight, respectively. Both methods should return a value in the range 0..1439.

'''
input : a string representing hh:mm
output : either minutes after mid_neight, or before midnight

Algo :
    Conver the input string to total minutes
        - split the string by  ':'
        - convert each array's string to numbers
        - assign hour and minutes  from the array
        - multiply 60 * hour + minutes
    After_midnight :
        The midnutes would be the number from before
    before_midnight is total number of minutes per day - the total minutes


'''


def after_midnight(string):
    array = string.split(':')
    hours, minutes = [int(ele) for ele in array]
    return (hours * 60 + minutes) % (60 * 24)

def before_midnight(string):
    return (60*24 - after_midnight(string) ) % (60 * 24)


print(after_midnight('00:00') == 0)
print(before_midnight('00:00') == 0)
print(after_midnight('12:34') == 754)
print(before_midnight('12:34') == 686)
print(after_midnight('24:00') == 0)
print(before_midnight('24:00') == 0)

File: js210_fundamentals_of_JS_Programming/objects/after_midnight2.js
const MINUTES_PER_HOUR = 60;
const HOURS_PER_DAY = 24;
const MINUTES_PER_DAY = HOURS_PER_DAY * MINUTES_PER_HOUR;
const DATE_PART = '1/1/2020';

function afterMidnight(timeStr) {
  let midnight = new Date(DATE_PART + ' 00:00');
  let time = new Date(DATE_PART + ' ' + timeStr);
  let hours = time.getHours();
  let minutes = time.getMinutes();
  return minutes + MINUTES_PER_HOUR * hours;
}

function beforeMidnight(timeStr) {
  let deltaMinutes = MINUTES_PER_DAY - afterMidnight(timeStr);
  if (deltaMinutes === MINUTES_PER_DAY) {
    deltaMinutes = 0;
  }

  return deltaMinutes;
}

console.log(afterMidnight('00:00')); // 0
console.log(beforeMidnight('00:00')); // 0
console.log(afterMidnight('12:34')); // 754
console.log(beforeMidnight('12:34')); // 686
